Realm/RLMRealm.mm

```diff
@@ -766,6 +766,32 @@ + (NSUInteger)schemaVersionAtPath:(NSString *)realmPath encryptionKey:(NSData *)
     return RLMRealmSchemaVersion(realm);
 }
 
++ (BOOL)migrationRequiredAtPath:(NSString *)realmPath encryptionKey:(NSData *)key error:(NSError **)outError {
+    RLMRealm *realm = RLMGetThreadLocalCachedRealmForPath(realmPath);
+    if (realm) {
+        // we've already successfully opened a realm, so a migration isn't necessary.
+        return NO;
+    } else {
+        NSError *error;
+        key = validatedKey(key) ?: keyForPath(realmPath);
+        realm = [[RLMRealm alloc] initWithPath:realmPath key:key readOnly:YES inMemory:NO dynamic:YES error:&error];
```

Comment:
No. The realm will be released at the end of the method, so it's safe to delete the file after calling this, as long as there are no other active realms. Although we should probably add a `+[RLMRealm isRealmOpenAtPath:]` method too, because otherwise it will be difficult for users to know for sure if they can delete the file.